docs/developer.md

```diff
@@ -0,0 +1,203 @@
+# Developer Docs
+
+In order to contribute to this project, make sure you have go 1.18 installed
+on your local machine.
+
+## Dependencies
+
+The build system assumes the following binaries are in the `PATH`:
+
+```
+make git go npm kind podman (or docker)
+```
+Once these tools are installed, run `make tools` to install all required
+project dependencies to ``tmp/bin``
+```
+make tools
+```
+## Development Environment Setup
+
+To setup your development, it is recommended to run the helper script
+`hack/kind/setup.sh`. This script does the following
+    * sets up a Kind cluster
+    * installs OLM
+    * sets up a local registry to push locally built operator and bundle images
+
+```sh
+./hack/kind/setup.sh
+```
+
+Once done, the cluster can be deleted by running
+
+```
+kind delete cluster --name obs-operator
+```
+
+## Running End to End tests
+
+Running E2E locally against the kind cluster that was setup following the
+instructions above is as easy as running `./test/run-e2e.sh`.
+
+```sh
+ ./test/run-e2e.sh
+```
+**NOTE:** `./test/run-e2e.sh --help` shows options that are useful when
+rerunning tests.
+
+
+# Running Operator In Kind Cluster
+
+
+Run `./hack/kind/setup.sh` to create a Kind cluster named `obs-operator` and
+to set up all the required dependencies (OLM, infra-node, registry) in the
+cluster.
+
+To cleanup the cluster or to retry the setup script after a failed attempt,
+run `kind delete cluster --name obs-operator`
+
+## Deploy ObO using `operator-sdk`
+
+Observability Operator relies heavily on (forked) Promethues Operator to do
+most of the heavy lifting of creation of Prometheus and Alertmanager.
+The easiest way to use deploy prometheus operator is to run the operator bundle
+and then scale the operator deployment to 0, so that the operator can be  run
+out of cluster using `go run`
```

Comment:
```suggestion
Observability Operator relies heavily on (forked) Promethues Operator to do
most of the heavy lifting of creation of Prometheus and Alertmanager.
The easiest way to use deploy prometheus operator is to run the `observability-operator`
bundle which installs `obo` and `prometheus-operator`,  and then scale the 
`observabilty-operator` deployment to 0, so that the operator can be  run
out of cluster using `go run`
```

Reply:
@JoaoBraveCoding how does this read?

Reply:
Perfect 👌 :chef_kiss: it way better